src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -234,6 +234,8 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
     register_cli_argument(scope, 'os_caching', options_list=['--storage-caching', '--os-disk-caching'], arg_group='Storage', help='Storage caching type for the VM OS disk.', **enum_choice_list([CachingTypes.read_only.value, CachingTypes.read_write.value]))
     register_cli_argument(scope, 'data_caching', options_list=['--data-disk-caching'], arg_group='Storage', help='Storage caching type for the VM data disk(s).', **enum_choice_list(CachingTypes))
 
+    register_cli_argument(scope, 'license_type', help='the license type if the Windows server image or disk being used was licensed on-premises')
```

Comment:
This is a good question. Till today, there is only one fixed value of "Windows_Server", but when CRP team rolls out the client OS image in near future, we will get a new license type "Windows_Client". I think for foreseeable future we can just use a choice list